src/clang-c-frontend/expr2ccode.cpp

```diff
@@ -0,0 +1,1509 @@
+    if(is_array_type(addr->ptr_obj->type))
+    {
+      array_type2tc arr_type = to_array_type(addr->ptr_obj->type);
+      // this arr_size is equal to the number of elements of the given array
+      expr2tc arr_size = arr_type->array_size;
+      greaterthan2tc gt(addr->ptr_obj, same->side_1);
+      greaterthan2tc gt2(
+        add2tc(same->side_1->type, same->side_1, gen_ulong(1)),
+        add2tc(arr_type->subtype, addr->ptr_obj, arr_size));
+      or2tc in_bounds(gt, gt2);
+      simplify(in_bounds);
+      return convert(migrate_expr_back(in_bounds), precedence);
+    }
+  }
+
+  equality2tc eq(same->side_1, same->side_2);
+  return convert(migrate_expr_back(not2tc(eq)), precedence);
+}
+
+std::string expr2ccodet::convert_pointer_offset(
+  const exprt &src,
+  unsigned &precedence [[maybe_unused]])
+{
+  std::string dest = "pointer_offset";
+  dest += '(';
+
+  forall_operands(it, src)
+  {
+    unsigned p;
+    std::string op = convert(*it, p);
+
+    if(it != src.operands().begin())
+      dest += ", ";
+
+    dest += op;
+  }
+
+  dest += ')';
+
+  return dest;
+}
+
+std::string expr2ccodet::convert_malloc(const exprt &src, unsigned &precedence)
+{
+  if(src.operands().size() != 1)
+    return convert_norep(src, precedence);
+
+  unsigned p0;
+  std::string op0 = convert(src.op0(), p0);
+
+  std::string dest = "malloc";
+  dest += '(';
+  dest += op0;
+  dest += ')';
+
+  return dest;
+}
+
+std::string expr2ccodet::convert_realloc(const exprt &src, unsigned &precedence)
+{
+  if(src.operands().size() != 1)
+    return convert_norep(src, precedence);
+
+  unsigned p0, p1;
+  std::string op0 = convert(src.op0(), p0);
+  std::string size = convert((const exprt &)src.cmt_size(), p1);
+
+  std::string dest = "realloc";
+  dest += '(';
+  dest += op0;
+  dest += ", ";
+  dest += size;
+  dest += ')';
+
+  return dest;
+}
+
+std::string expr2ccodet::convert_alloca(const exprt &src, unsigned &precedence)
+{
+  if(src.operands().size() != 1)
+    return convert_norep(src, precedence);
+
+  unsigned p0;
+  std::string op0 = convert(src.op0(), p0);
+
+  std::string dest = "alloca";
+  dest += '(';
+  dest += op0;
+  dest += ')';
+
+  return dest;
+}
+
+std::string expr2ccodet::convert_nondet(const exprt &src, unsigned &precedence)
+{
+  if(src.operands().size() != 0)
+    return convert_norep(src, precedence);
+
+  std::string type_str = "";
+
+  if(src.type().is_bool())
+  {
+    type_str += "bool";
+  }
+  else if(src.type().id() == "signedbv" || src.type().id() == "unsignedbv")
+  {
+    BigInt width = string2integer(src.type().width().as_string());
+
+    if(src.type().id() == "unsignedbv")
+      type_str += "u";
+
+    if(width == config.ansi_c.int_width)
+      type_str += "int";
+
+    if(width == config.ansi_c.long_int_width)
+      type_str += "long";
+
+    if(width == config.ansi_c.char_width)
+      type_str += "char";
+
+    if(width == config.ansi_c.short_int_width)
+      type_str += "short";
```

Comment:
Is there a `long long` version of `nondet`? What about other bit-widths, can/do they occur?

Reply:
Those seem to be the only cases of `nondet`s that appear in `esbmc_intrinsics.h`. Apart from `__VERIFIER_nondet_schar()` which never appears in the regression, and which seems to be the same as `__VERIFIER_nonder_char()`